india_compliance/gst_india/report/gstr_1/gstr_1.py

```diff
@@ -557,6 +561,114 @@ def get_items_based_on_tax_rate(self):
+                Sum(
+                    Case()
+                    .when(
+                        taxes.account_head == igst_account,
+                        taxes.tax_amount,
+                    )
+                    .else_(0)
+                ).as_("total_igst_amount"),
+                Sum(
+                    Case()
+                    .when(
+                        taxes.account_head == cgst_account,
+                        taxes.tax_amount,
+                    )
+                    .else_(0)
+                ).as_("total_cgst_amount"),
+                Sum(
+                    Case()
+                    .when(
+                        taxes.account_head == sgst_account,
+                        taxes.tax_amount,
+                    )
+                    .else_(0)
+                ).as_("total_sgst_amount"),
+                (
+                    Sum(
+                        Case()
+                        .when(
+                            taxes.account_head == cess_account,
+                            taxes.tax_amount,
+                        )
+                        .else_(0)
+                    )
+                    + Sum(
+                        Case()
+                        .when(
+                            taxes.account_head == cess_non_advol,
+                            taxes.tax_amount,
+                        )
+                        .else_(0)
+                    )
+                ).as_("total_cess_amount"),
+            )
+            .groupby(taxes.parent)
+        )
+
+        query = (
+            frappe.qb.from_(si)
+            .left_join(taxable_value_query)
+            .on(si.name == taxable_value_query.parent)
+            .left_join(taxes_query)
+            .on(si.name == taxes_query.parent)
+            .select(
+                si.ecommerce_gstin,
+                Sum(IfNull(taxable_value_query.total_taxable_value, 0)).as_(
+                    "total_taxable_value"
+                ),
+                Sum(IfNull(taxes_query.total_igst_amount, 0)).as_("total_igst_amount"),
+                Sum(IfNull(taxes_query.total_cgst_amount, 0)).as_("total_cgst_amount"),
+                Sum(IfNull(taxes_query.total_sgst_amount, 0)).as_("total_sgst_amount"),
+                Sum(IfNull(taxes_query.total_cess_amount, 0)).as_("total_cess_amount"),
+                Case()
+                .when(si.is_reverse_charge == 1, SUPECOM.US_9_5.value)
+                .else_(SUPECOM.US_52.value)
+                .as_("ecommerce_supply_type"),
+            )
+            .where(si.is_opening == "No")
+            .where(si.docstatus == 1)
+            .where(IfNull(si.ecommerce_gstin, "") != "")
+            .where(IfNull(si.billing_address_gstin, "") != si.company_gstin)
+            .where(
+                Date(si.posting_date).between(
+                    self.filters.from_date, self.filters.to_date
+                )
+            )
+            .where(si.company == self.filters.company)
```

Comment:
Subqueries can be with these same conditions. This will reduce the data returned by the subquery.